public/electron.js

```diff
@@ -1,96 +1,123 @@
-const { app, BrowserWindow, dialog, globalShortcut, Menu } = require('electron')
-const path = require('path')
-const redis = require('redis')
-const url = require('url')
-
-const util = require('util')
-const childProcess = require('child_process')
-const execFile = util.promisify(childProcess.execFile)
-
-const logger = require('./logger')
-const preferences = require('./preferences')
-const updater = require('./updater')
-const { getBin } = require('./util')
-
-let mainWindow
-let queues
-
-async function startRedis() {
-  const redisPort = preferences.get('general.redisPort')
-  await execFile(
-    getBin('redis-server'),
-    [
-      '--daemonize', 'yes',
-      '--port', redisPort,
-      '--appendonly', 'yes',
-      '--appendfsync', 'everysec'
-    ],
-    { cwd: app.getPath('userData') }
-  )
-
-  return new Promise((resolve, reject) => {
-    const client = redis.createClient({
-      port: redisPort,
-      retry_strategy(options) {
-        if (options.error && options.error.code === 'ECONNREFUSED') {
-          logger.error(options.error)
-        }
+const { app, BrowserWindow, globalShortcut, Menu } = require('electron')
 
-        if (options.attempt > 3) {
-          const err = new Error(
-            `Are you sure port ${redisPort} is available? Try killing any ` +
-            `process listening on port ${redisPort} or changing redis port ` +
-            'in Preferences, restart publish50, and try again!'
-          )
+app.on('ready', async () => {
 
-          reject(err)
-          return err
+  const path = require('path')
+  const redis = require('redis')
+  const url = require('url')
+
+  const util = require('util')
+  const childProcess = require('child_process')
+  const execFile = util.promisify(childProcess.execFile)
+
+  const logger = require('./logger')
+  const preferences = require('./preferences')
+  const updater = require('./updater')
+  const { getBin } = require('./util')
+  const _dialog = require('./dialog.js')
```

Comment:
Since we're not using electron's dialog module in this module, let's rename this to `dialog` and only name it `dialog_` in case electron's dialog is required in the same module.